### PR TITLE
Generate a pmtiles layer of schools and hospitals for the scheme browse page

### PR DESCRIPTION
**Pull request by dabreegster**
One easy layer inspections has requested for the scheme-browse page is key trip generators. This PR starts with primary+secondary schools and hospitals. Comments elsewhere in the PR about technical decisions. I'm about to open another PR in the ATIP repo for adding these to the frontend.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -59,3 +59,34 @@ IMPORTANT! The `importer` crate's dependency on `osm2streets` must be kept in
...
+ATIP loads extra layers showing key trip generators. These layers are
+England-wide, rather than being split into a file per area, because they're
+being used on the country-wide scheme browse page. Each layer is a single
+[PMTiles](https://protomaps.com/docs/pmtiles/) file.
+
+To run this:
+
+1.  Get `england-latest.osm.pbf` from Geofabrik. The `split_uk_osm.sh` script above does this.
+2.  Install [osmium](https://osmcode.org/osmium-tool)
+3.  Install [tippecanoe](https://github.com/felt/tippecanoe) for transforming GeoJSON to PMTiles
```
Comment (dabreegster) [inline README.md:74]: Note another dependency here. I had no trouble installing it locally. Right now I'm the only one running things in this reop, but happy to work on ensuring there's a Docker image that can do everything too.

File: README.md [hunk 2]
```diff
@@ -59,3 +59,34 @@ IMPORTANT! The `importer` crate's dependency on `osm2streets` must be kept in
...
+4.  Run `cd pois; generate_pois.py ../england-latest.osm.pbf`
+5.  Pick an arbitrary version number, and upload the file: `aws s3 cp --dry *.pmtiles s3://atip.uk/layers/v1/`
+
+You can debug a PMTiles file using <https://protomaps.github.io/PMTiles>.
+
+### One-time cloud setup for PMTiles
+
+Currently we're using S3 and Cloudfront to host files generated by this repo.
+As a one-time setup, the S3 CORS policy has to be edited per
+<https://protomaps.com/docs/pmtiles/cloud-storage>.
```
Comment (dabreegster) [inline README.md:84]: From these docs, I don't anticipate any complications hosting the files on GCP

File: pois/generate_pois.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+# This tool takes an England-wide osm.pbf (from
+# http://download.geofabrik.de/europe/great-britain/england-latest.osm.pbf) and
+# generates files with point-of-interest layers for ATIP.
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("input", help="Path to england-latest.osm.pbf file", type=str)
+    args = parser.parse_args()
+
+    # https://wiki.openstreetmap.org/wiki/Tag:amenity%3Dschool indicates
```
Comment (dabreegster) [inline pois/generate_pois.py:17]: We can share this initial version for feedback to see if universities and other similar facilities should be in the same layer or not

File: pois/generate_pois.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    # primary and secondary schools
+    generatePolygonAmenity(args, "school", "schools")
+
+    # Note https://wiki.openstreetmap.org/wiki/Tag:amenity%3Dhospital doesn't
+    # cover all types of medical facility
+    generatePolygonAmenity(args, "hospital", "hospitals")
+
+
+# Extract `amenity={amenity}` polygons from OSM, and only keep a name attribute.
```
Comment (dabreegster) [inline pois/generate_pois.py:26]: I thought about exposing a finer granularity of individual buildings constituting a hospital/school/etc, but for the purposes of understanding schemes in context of trip generators, that's not useful information. This tag is drawn around the entire campus / area. See https://www.openstreetmap.org/way/175944466 for an example

File: pois/generate_pois.py [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    # Remove files from any previous run
+    try:
+        os.remove(f"{filename}.osm.pbf")
+        os.remove(f"{filename}.geojson")
+        os.remove(f"{filename}.pmtiles")
+    except:
+        pass
+
+    # First extract a .osm.pbf with all amenity={name} features
+    # TODO Do we need nwr? We don't want points further on
```
Comment (dabreegster) [inline pois/generate_pois.py:37]: nwr is nodes, ways, and relations. Below we filter for just polygons. Doesn't really hurt anything to extract all 3 here

File: pois/generate_pois.py [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+
+
+# For each GeoJSON feature, keep only the name attribute. Overwrites the given file.
+def remove_extra_properties(path):
+    print(f"Removing extra properties from {path}")
+    gj = {}
+    with open(path) as f:
+        gj = json.load(f)
+        for feature in gj["features"]:
+            # Remove all properties except for "name"
```
Comment (dabreegster) [inline pois/generate_pois.py:80]: Will ask for feedback to see what else is useful to surface. Maybe just an OSM ID, letting people click to open and see all other attributes.

File: pois/generate_pois.py [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+        os.remove(f"{filename}.geojson")
+        os.remove(f"{filename}.pmtiles")
+    except:
+        pass
+
+    # First extract a .osm.pbf with all amenity={name} features
+    # TODO Do we need nwr? We don't want points further on
+    run(
+        [
+            "osmium",
```
Comment (dabreegster) [inline pois/generate_pois.py:40]: There's a variety of toolchains and libraries in different languages for extracting OSM data. For simple cases of grabbing geometric objects matching some tags, osmium is dirt simple and extremely performant

File: README.md [hunk 8]
```diff
@@ -59,3 +59,34 @@ IMPORTANT! The `importer` crate's dependency on `osm2streets` must be kept in
 sync with the git version used by ATIP's [route info
 crate](https://github.com/acteng/atip/tree/map_model/route_info). Otherwise,
 the binary file format may be incompatible. Use `cargo update -p osm2streets`.
+
+## POIs (points of interest)
+
+ATIP loads extra layers showing key trip generators. These layers are
+England-wide, rather than being split into a file per area, because they're
+being used on the country-wide scheme browse page. Each layer is a single
+[PMTiles](https://protomaps.com/docs/pmtiles/) file.
```
Comment (dabreegster) [inline README.md:68]: The pmtiles docs are worth skimming through for more background. In short, we're using them as a much more performant variation of a .geojson file. We don't want to load the entire England-wide dataset, just the areas that the user zooms in on. Vector tiles are the general technology that achieve this. Serving vector tiles is kind of tedious, and pmtiles are a newer approach that's simple and fast. They're packed into a single file, and the browser uses HTTP range requests to grab only the necessary parts.

**Review by dabreegster — COMMENTED** [review]
CC @ADD-William-WaltersDavis in case y'all're interested in having simple toggleable layers to show major trip destinations. This is a simple example how to extract the data from OSM, and very compactly store+serve in pmtiles

File: pois/generate_pois.py (hunk 6 above)
Comment (Pete-Y-CS) [reply pois/generate_pois.py:80]: This sounds nice

**Review by Pete-Y-CS — APPROVED** [review]

**Pull request merged**